src/test/app/PermissionedDomains_test.cpp

```diff
@@ -0,0 +1,557 @@
+
+        // Make sure owner count reflects the existing domain.
+        BEAST_EXPECT(env.ownerCount(alice) == 1);
+        auto const objID = pdomain::getObjects(alice, env).begin()->first;
+        BEAST_EXPECT(pdomain::objectExists(objID, env));
+
+        // Delete domain that belongs to user.
+        env(pdomain::deleteTx(alice, domain));
+        auto const tx = env.tx()->getJson(JsonOptions::none);
+        BEAST_EXPECT(tx[jss::TransactionType] == "PermissionedDomainDelete");
+
+        // Make sure the owner count goes back to 0.
+        BEAST_EXPECT(env.ownerCount(alice) == 0);
+
+        // The object needs to be gone.
+        BEAST_EXPECT(pdomain::getObjects(alice, env).empty());
+        BEAST_EXPECT(!pdomain::objectExists(objID, env));
+    }
+
+    void
+    testAccountReserve()
+    {
+        // Verify that the reserve behaves as expected for minting.
+        testcase("Account Reserve");
+
+        using namespace test::jtx;
+
+        Env env(*this, withFeature_);
+        Account const alice("alice");
+
+        // Fund alice enough to exist, but not enough to meet
+        // the reserve.
+        auto const acctReserve = env.current()->fees().accountReserve(0);
+        auto const incReserve = env.current()->fees().increment;
+        env.fund(acctReserve, alice);
+        env.close();
+        BEAST_EXPECT(env.balance(alice) == acctReserve);
+        BEAST_EXPECT(env.ownerCount(alice) == 0);
+
+        // alice does not have enough XRP to cover the reserve.
+        pdomain::Credentials credentials{{alice, "first credential"}};
+        env(pdomain::setTx(alice, credentials), ter(tecINSUFFICIENT_RESERVE));
+        BEAST_EXPECT(env.ownerCount(alice) == 0);
+        BEAST_EXPECT(pdomain::getObjects(alice, env).size() == 0);
+        env.close();
+
+        auto const baseFee = env.current()->fees().base.drops();
+
+        // Pay alice almost enough to make the reserve.
+        env(pay(env.master, alice, incReserve + drops(2 * baseFee) - drops(1)));
+        BEAST_EXPECT(
+            env.balance(alice) ==
+            acctReserve + incReserve + drops(baseFee) - drops(1));
+        env.close();
+
+        // alice still does not have enough XRP for the reserve.
+        env(pdomain::setTx(alice, credentials), ter(tecINSUFFICIENT_RESERVE));
+        env.close();
+        BEAST_EXPECT(env.ownerCount(alice) == 0);
+
+        // Pay alice enough to make the reserve.
+        env(pay(env.master, alice, drops(baseFee) + drops(1)));
+        env.close();
+
+        // Now alice can create a PermissionedDomain.
+        env(pdomain::setTx(alice, credentials));
+        env.close();
+        BEAST_EXPECT(env.ownerCount(alice) == 1);
+    }
+
+public:
+    void
+    run() override
+    {
+        testEnabled();
+        testDisabled();
+        testSet();
+        testDelete();
+        testAccountReserve();
+    }
+};
+
+BEAST_DEFINE_TESTSUITE_PRIO(PermissionedDomains, app, ripple, 2);
```

Comment:
Why does this test suite need a different priority?

Reply:
removed